### PR TITLE
feat(teams): Allow members to request to add members to teams

### PR DESCRIPTION
**Pull request by maheskett**
`Open Membership` team permissions were recently changed to allow members to add other members to teams, without requiring team admin permissions #16244

This is a follow-up PR to allow members to *request* to add other members to their teams, if they don't have permission and `Open Membership` is disabled.

One other change has been to remove role comparison (i.e. "owner" can't add a "manager" to a team). This simplifies team permissions and access requests, and fixes a bug where role "admins" could add an "owner" to a team but "managers" could not.

### CODE REVIEW

File: src/sentry/static/sentry/app/components/modals/teamAccessRequestModal.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    return (
+      <React.Fragment>
+        <Body>
+          {t(
+            'You do not have permission to add members to teams, but we will send a request to your organization admins for approval.'
+          )}
+        </Body>
+        <Footer>
+          <ButtonGroup>
+            <Button onClick={closeModal} autoFocus>
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/modals/teamAccessRequestModal.tsx]: Should both buttons have `autoFocus`?

File: src/sentry/static/sentry/app/components/modals/teamAccessRequestModal.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      await api.requestPromise(
+        `/organizations/${orgId}/members/${memberId}/teams/${teamId}/`,
+        {
+          method: 'POST',
+        }
+      );
+      this.setState({createBusy: false});
+      addSuccessMessage(t('Team request sent for approval'));
+    } catch (err) {
+      this.setState({createBusy: false});
```
Comment (EvanPurkhiser) [inline src/sentry/static/sentry/app/components/modals/teamAccessRequestModal.tsx]: I think you can pull this setState out to after the try/catch and just have it once.

**Review by evanpurkhiser — APPROVED** [review]
Looks good to me, great addition.

**Review by billyvg — APPROVED** [review]
Thanks for all the tests!

**Pull request merged**